core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -164,35 +164,26 @@ sealed trait Chunk[+A] extends ChunkLike[A] { self =>
   }
 
   def dropWhileM[R, E](p: A => ZIO[R, E, Boolean]): ZIO[R, E, Chunk[A]] = {
-    val len                                       = self.length
-    var dest: ZIO[R, E, (Boolean, Int, Array[A])] = UIO.succeedNow((true, 0, null.asInstanceOf[Array[A]]))
+    val len     = self.length
```

Comment:
This whole function body needs to be encased in `ZIO.effectSuspendTotal`. Right now, it is not referentially transparent, which will cause bugs.

Reply:
Thanks! I got it.
Should I also encase the function bodies of `mapAccumM` in `ZIO.effectSuspendTotal`?

Reply:
Yes please, the construction of a builder is impure, and those methods must be wrapped in `effectSuspendTotal`. Thank you so much! 🙏